ext/ddtrace_profiling_native_extension/collectors_cpu_and_wall_time_worker.c

```diff
@@ -89,13 +90,13 @@ struct cpu_and_wall_time_worker_state {
   bool gc_profiling_enabled;
   bool no_signals_workaround_enabled;
   bool dynamic_sampling_rate_enabled;
-  int allocation_sample_every;
   bool allocation_profiling_enabled;
   VALUE self_instance;
   VALUE thread_context_collector_instance;
   VALUE idle_sampling_helper_instance;
   VALUE owner_thread;
-  dynamic_sampling_rate_state dynamic_sampling_rate;
+  dynamic_sampling_rate_state cpu_dynamic_sampling_rate;
+  discrete_dynamic_sampler *allocation_sampler;
```

Comment:
Yeah I tend to prefer explicit initialization rather than magically-working 0-valued objects but I can definitely preserve consistency here.

Reply:
I mostly did it to avoid having to deal with the memory allocations. You're very right on the initialization point: the pointer version forces explicit initialization, whereas on this version we need to remember to initialize.

I still think it's worth being consistent, so if you prefer the malloc version, do you mind switching the existing one to that?

Reply:
Migrated mine to follow suit to see how I felt about it. Works alright but it bothers me that it forces us to move the struct definition to the header, potentially allowing users to directly affect the internal state. Currently users is just us 2 so I can live with it but that's the only con I could come up with 😂

Reply:
Arghhh C is so awkward lol.